File: java/rna-transcription/src/main/java/RnaTranscription.java
public class RnaTranscription {

    public Character dnaToRna(Character c) {
      switch(c) {
        case 'G': return 'C';
        case 'C': return 'G';
        case 'T': return 'A';
        case 'A': return 'U';
        default:
          throw new IllegalArgumentException("Invalid DNA nucleotide: " + c);
      }
    }

    public String transcribe(String dnaStrand) {
      return dnaStrand.chars()
                      .map(c -> dnaToRna((char)c))
                      .collect(StringBuilder::new,
                               StringBuilder::appendCodePoint,
                               StringBuilder::append)
                      .toString();
    }
}

File: php/rna-transcription/rna-transcription.php
<?php

function toRna($dna_strand) {
  $transcribe = function($nucleotide) {
    $dna_to_rna = ['G' => 'C', 'C' => 'G', 'T' => 'A', 'A' => 'U'];
    return $dna_to_rna[$nucleotide];
  };

  $dna_nucleotides = str_split($dna_strand);
  $rna_nucleotides = array_map($transcribe, $dna_nucleotides);
  return implode($rna_nucleotides);
}
